### DISCUSSION TITLE
How to handle certificate browser popup dialogues?

### DISCUSSION THREAD

**Discussion by MihhailShramov** [selenide/selenide#2090] [Q&A]
Hello all,

I faced an issue with the certificate popup dialogue in my project.
Website before loading the page is asking to confirm the cert, is it somehow possible to accept such popup using Selenide?

I have already tried to make it with **switchTo().alert().accept();** and **Selenide.actions().sendKeys(Keys.ENTER).build().perform();** but looks like my code was even not reached when **open(URL);** was executed.

When **open();** method is used selenide waits until page is rendered and no actions can be performed during this time.

So tests are giving this error **org.openqa.selenium.TimeoutException: timeout: Timed out receiving message from renderer: -0.006**

Thank you in advance.

<img width="408" alt="image" src="https://user-images.githubusercontent.com/98530531/209309751-934aaf96-1dd3-4484-99fc-7347b9e176a7.png">

**Comment by asolntsev** [comment]
Hi @MihhailShramov ,
afaik it's not possible to close this dialog by Selenium/Selenide.
Webdriver doesn't provide a mechanism to access native dialogs in browsers.

You probably need to investigate if there are some webdriver options you could pass when opening the browser. Probably there are some options like "accept all certs"...

**Comment by MihhailShramov** [comment]
Hi @asolntsev
I have tried to pass some arguments to webdriver and only **"--ignore-urlfetcher-cert-requests"** makes some visible effect on popup.
Whatever it is still not what I required, as this argument just closes the pop up, but I need to accept it.

Also regarding the accepting the certificates, I have found **chromeOptions.setAcceptInsecureCerts(true);** but it also not makes much sense as not making any effect to pop up.

Maybe there are some more arguments to try, but everything I looked in google, makes not much progress.

**Reply by MihhailShramov** [reply]
A little update, seems I found a "solution" to modify the **HKEY_CURRENT_USER\SOFTWARE\Policies\Google\Chrome\AutoSelectCertificateForUrls** in **regedit** by adding there value with **name: 1** and **value: {"pattern":"[*.]","filter":{}}**.

This way is forcing Chrome to always accept the certificate pop up, but now the question how good and reliable approach it is.
